src/lib/TransactionSender.ts

```diff
@@ -128,8 +130,13 @@ export class TransactionSender {
   }
 
   async getChainId() {
+    console.info('getChainId', this.chainId);
     if (this.chainId === undefined || this.chainId === null) {
-      this.chainId = Number(await this.client.eth.net.getId());
+      this.logger.error('chainId is undefined, trying to get it from the node');
+      this.chainId = Number(await this.client.eth.getChainId());
+    }
+    if (this.chainId === undefined || this.chainId === null) {
```

Comment:
did you test this? if you're unable to get the chain id from the node Number(null) is 0 and Number(undefined) is NaN and then if you evaluate that condition with 0 or NaN it is false in both cases so you never throw the error